routes/projects.js

```diff
@@ -1,9 +1,47 @@
 var express = require('express');
 var router = express.Router();
+var request = require('request')
+
+// Load up the environment variables.
+require('dotenv').config()
 
 /* GET projects listing. */
-router.get('/', function(req, res, next) {
-  res.send('respond with a resource');
+router.get('/:user', function(req, res, next) {
+
+  // User info from keycloak
+  const user = req.params.user;
+
+  // Login with admin credentials
+  var username = process.env.USERNAME,
+  password = process.env.PASSWORD,
+  url = "https://" + username + ":" + password + "@bnc.brown.edu/xnat-dev";
+
+  // Get user accesss token for the specific username from keycloak using XNAT User Alias Token API
+  request(
+    {
+        url : url + '/data/services/tokens/issue/user/' + user
+    },
+    function (error, response, body) {
+      var token = JSON.parse(body)
+      
+      // Login with user credentials
+      var username = token.alias,
+      password = token.secret,
+      url = "https://" + username + ":" + password + "@bnc.brown.edu/xnat-dev";
+
+      // Get the projects for that specific user using XNAT Project API
+      request(
+        {
+            url : url + '/data/projects'
+        },
+        function (error, response, body) {
+          res.send(body)
+        }
+      );
+    }
+  );
```

Comment:
Is the connection closed? I thought it was necessary to do a DELETE request to `/data/JSESSION` to close the session. See https://wiki.xnat.org/display/XKB/Session+management+in+XNAT+scripts
@broarr you may have better insight into that since you fixed that for `XNATs-tools`

Reply:
For closing the user session maybe doing a get `/data/JSESSION` and then a delete request would work? For admin I am manually getting the token for my account and setting it in the env, which I don't delete the session anywhere.

Reply:
```
// Get the projects for that specific user using XNAT Project API
      request(
        {
            url : url + '/data/projects'
        },
        function (error, response, body) {
          request(
            {
                url : url + '/data/JSESSION'
            },
            function (error, response, body) {
              request.delete(
                {
                    url : url + '/data/JSESSION'
                }
              );
            }
          );
          res.send(body)
        }
      );
```
this is working! @broarr not sure if it good to have many nested requests, let me know if there is better way for doing this

Reply:
I'm not sure. It's pretty easy to observe though, run this code twice and then log into `xnat-dev` as the admin and see how many active sessions are open. If it's two we need to think about the session management stuff. I don't know if sessions are handled differently with tokens. I should have checked that....

Reply:
> ```
> // Get the projects for that specific user using XNAT Project API
>       request(
>         {
>             url : url + '/data/projects'
>         },
>         function (error, response, body) {
>           request(
>             {
>                 url : url + '/data/JSESSION'
>             },
>             function (error, response, body) {
>               request.delete(
>                 {
>                     url : url + '/data/JSESSION'
>                 }
>               );
>             }
>           );
>           res.send(body)
>         }
>       );
> ```
> 
> this is working! @broarr not sure if it good to have many nested requests, let me know if there is better way for doing this

Oh funny, I must have had an old tab open without realizing it. Sorry for the out of order thread! 😞 

`async`/`await` will really flatten your javascript out. I think that's probably the best way to deal with this many nested callbacks. I made a comment about that below

Reply:
@Rashi1997 - Have you asserted that your code works by logging into xnat-dev? I think you need to supply the token as an argument to delete, but I don't remember off the top of my head